### PR TITLE
Issue 404 - Separate NUnitLite and Framework Tests

### PR DESCRIPTION
**Pull request by CharliePoole**
This separates the nunitlite tests into their own assembly, which is itself executable using nunitlite.

Additionally, I reorganized some directories and namespaces, reflecting the fact that nunitlite is now entirely a runner and no longer needs a Runner directory as it did when it was in the framework.

### CODE REVIEW

**Comment by CharliePoole** [comment]
Don't merge this yet - I have to fix the CF build.

**Comment by CharliePoole** [comment]
Ready to merge now.

File: nunit.sln [hunk 1]
```diff
@@ -1,8 +1,6 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 14
-VisualStudioVersion = 14.0.23107.0
-MinimumVisualStudioVersion = 10.0.40219.1
+# Visual Studio 2012
```
Comment (rprouse) [inline nunit.sln]: Still using VS 2012? Charlie, we need to talk :smile:

File: src/NUnitFramework/framework/Internal/Filters/CompositeFilter.cs [hunk 2]
```diff
@@ -57,15 +55,12 @@ public CompositeFilter( params ITestFilter[] filters )
...
 
         /// <summary>
-        /// Return an array of the composing filters
+        /// Return a list of the composing filters.
         /// </summary>
-        public ITestFilter[] Filters
-        {
-            get { return _filters.ToArray(); }
-        }
+        public IList<ITestFilter> Filters { get; private set; }
```
Comment (rprouse) [inline src/NUnitFramework/framework/Internal/Filters/CompositeFilter.cs]: I've been meaning to do this, that build warning was annoying me :+1:

**Comment by rprouse** [comment]
This looks good, but I expect I will need to update the DNX solution with the same changes. I will pull this branch and make those changes.

File: nunit.sln (hunk 1 above)
Comment (CharliePoole) [inline nunit.sln]: There are things I don't like about 2013, so I use 2012 for NUnit and only use 2013 for the adapter, which requires it.

We should aim to have our own releases constantly improve on their predecessors, unlike some other folks... ;-)

Comment (CharliePoole) [inline nunit.sln]: With a bit more seriousness: I have always tried to keep using the lowest version of the IDE that will work for building NUnit. That keeps it working for others who may want to contribute but can't afford Microsoft prices. The VS2015 community edition may change my approach though.

Comment (rprouse) [inline nunit.sln]: Personally, I really like 2015 and the Community Edition is a great
offering. Besides, if we are going to support newer platforms, you are
going to have to move ;-)

On 20 October 2015 at 13:55, CharliePoole <notifications@github.com> wrote:

> In nunit.sln
> <https://github.com/nunit/nunit/pull/909#discussion_r42529966>:
>
> > @@ -1,8 +1,6 @@
> >  ﻿
> >  Microsoft Visual Studio Solution File, Format Version 12.00
> > -# Visual Studio 14
> > -VisualStudioVersion = 14.0.23107.0
> > -MinimumVisualStudioVersion = 10.0.40219.1
> > +# Visual Studio 2012
>
> With a bit more seriousness: I have always tried to keep using the lowest
> version of the IDE that will work for building NUnit. That keeps it working
> for others who may want to contribute but can't afford Microsoft prices.
> The VS2015 community edition may change my approach though.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nunit/nunit/pull/909/files#r42529966>.
>



-- 

Rob Prouse



I welcome VSRE emails. Learn more at http://vsre.info/

**Comment by CharliePoole** [comment]
I decided to leave the DNX solution for you. :-) We'll want to change that and make it mainstream at some point. I'm actually not clear if I need Windows 10 to build it - something for inclusion in issue #240.

**Comment by CharliePoole** [comment]
Do you want to add your changes to this branch before we merge?

File: nunit.sln (hunk 1 above)
Comment (CharliePoole) [inline nunit.sln]: Me too. But of course my real plan is to just move out. Then I can drop windows entirely and stick with Linux. :-)

**Comment by rprouse** [comment]
I am making the changes on that branch now. I will upload them and merge
when I am done.

You don't need Windows 10 to build the DNX solution, but you do need VS
2015 with the Windows 10 SDK installed and .Net Core needs to be updated to
Beta8, although I think that happens automatically. I will need to test on
a clean machine and document.

On 20 October 2015 at 14:01, CharliePoole <notifications@github.com> wrote:

> Do you want to add your changes to this branch before we merge?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nunit/nunit/pull/909#issuecomment-149648840>.
>



-- 

Rob Prouse



I welcome VSRE emails. Learn more at http://vsre.info/

**Comment by CharliePoole** [comment]
That's about what I assumed. Since you need the SDK, it sounds like I would not be able to run tests without Windows 10.

**Comment by rprouse** [comment]
No, I managed to run the tests on Windows 8.1. I only have a DNX runner at
the moment, which is a .NET core console application. It requires some DNX
knowledge to get running, but is easy once you figure it out. Should run on
Linux too :-)

On 20 October 2015 at 14:11, CharliePoole <notifications@github.com> wrote:

> That's about what I assumed. Since you need the SDK, it sounds like I
> would not be able to run tests without Windows 10.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nunit/nunit/pull/909#issuecomment-149651379>.
>



-- 

Rob Prouse



I welcome VSRE emails. Learn more at http://vsre.info/

**Pull request merged**